app/uk/gov/hmrc/eusubsidycompliancefrontend/services/AuditService.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eusubsidycompliancefrontend.services
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.Logging
+import play.api.libs.json.{Json, Writes}
+import play.api.mvc.Request
+import uk.gov.hmrc.eusubsidycompliancefrontend.models.AuditEvent
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions.auditHeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+
+@ImplementedBy(classOf[AuditServiceImpl])
+trait AuditService {
+
+  def sendEvent[A <: AuditEvent](auditEvent: A)(implicit
+        hc: HeaderCarrier,
+        writes: Writes[A],
+        request: Request[_]
+  ): Unit
+
+}
+
+@Singleton
+class AuditServiceImpl @Inject() (auditConnector: AuditConnector)(implicit ec: ExecutionContext)
+  extends AuditService
+    with Logging {
+
+  private val auditSource: String = "eu-subsidy-compliance-frontend"
+
+  override def sendEvent[A <: AuditEvent](
+    auditEvent: A
+  )(implicit
+       hc: HeaderCarrier,
+       writes: Writes[A],
+       request: Request[_]
+   ): Unit = {
+    val extendedDataEvent = ExtendedDataEvent(
+      auditSource = auditSource,
+      auditType = auditEvent.auditType,
+      detail = Json.toJson(auditEvent),
+      tags = hc.toAuditTags(auditEvent.transactionName, request.uri)
+    )
+    auditConnector.sendExtendedEvent(extendedDataEvent).onComplete {
+      case Success(_) => ()
+      case Failure(e) =>
+        // WARNING: do not change log message here without updating alert-config as well
+        logger.warn(s"Could not audit ${auditEvent.auditType} event: ${e.getMessage}")
```

Comment:
If the format matters so much perhaps extract this to a separate audit logging object that provides helper methods.

Reply:
I thought its just one line and it's ok if we write that in logger in the file.

Reply:
Ahh right, I wasn't sure if we were going to do this in several other places. The danger with leaving it like this is that it's really easy to break. If we are going to do this in several places then we should consider introducing a helper to encapsulate that logging logic, and it'll make it far less likely for someone to come along in the future and break it without realising how important it is. A `WARNING - do not change` comment is definitley a sign that there's a better way to structure the code. ;)